CRM/Core/ClassLoader.php

```diff
@@ -104,7 +104,8 @@ public function register($prepend = FALSE) {
     }
     $civicrm_base_path = dirname(dirname(__DIR__));
 
-    require_once dirname(dirname(__DIR__)) . '/vendor/autoload.php';
+    // @todo: need some way to check if we should register vendor or not
```

Comment:
It's possible to check if a required class exists, and if it does not then load the autoloader?

We could check if `\Composer\Autoload\ClassLoader` exists. In the generated `autoloader_real.php` the randomly named class has

```
    public static function loadClassLoader($class)
    {
        if ('Composer\Autoload\ClassLoader' === $class) {
            require __DIR__ . '/ClassLoader.php';
        }
    }
```

So if that exists, we know an autoloader has been bootstrapped.